internal/controller/dnsrecord_controller.go

```diff
@@ -237,18 +271,34 @@ func (r *DNSRecordReconciler) SetupWithManager(mgr ctrl.Manager, maxRequeue, val
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.DNSRecord{}).
-		Watches(&v1alpha1.ManagedZone{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o client.Object) []reconcile.Request {
+		Watches(&v1.Secret{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o client.Object) []reconcile.Request {
 			logger := log.FromContext(ctx)
-			toReconcile := []reconcile.Request{}
-			// list dns records in the maanagedzone namespace as they will be in the same namespace as the zone
+			var toReconcile []reconcile.Request
+
 			records := &v1alpha1.DNSRecordList{}
 			if err := mgr.GetClient().List(ctx, records, &client.ListOptions{Namespace: o.GetNamespace()}); err != nil {
-				logger.Error(err, "failed to list dnsrecords ", "namespace", o.GetNamespace())
+				logger.Error(err, "failed to list records ", "namespace", o.GetNamespace())
 				return toReconcile
 			}
 			for _, record := range records.Items {
-				if record.Spec.ManagedZoneRef.Name == o.GetName() {
-					logger.Info("managed zone updated", "managedzone", o.GetNamespace()+"/"+o.GetName(), "enqueuing dnsrecord ", record.GetName())
+				// get the zone for this record
+				managedZone := &v1alpha1.ManagedZone{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      record.Spec.ManagedZoneRef.Name,
+						Namespace: record.Namespace,
+					},
+				}
+				err := r.Get(ctx, client.ObjectKeyFromObject(managedZone), managedZone, &client.GetOptions{})
+				if err != nil {
+					continue
+				}
+				// if secret is related to zone for this record, add this record to requeue items
+				if managedZone.Spec.SecretRef.Name == o.GetName() {
+					logger.Info(
+						"managed zone Secret updated",
+						"managedZone", managedZone.Namespace+"/"+managedZone.Name,
+						"secret", o.GetNamespace()+"/"+o.GetName(),
+						"enqueuing record ", record.GetName())
 					toReconcile = append(toReconcile, reconcile.Request{NamespacedName: client.ObjectKeyFromObject(&record)})
 				}
```

Comment:
Why is the watch for a secret on the dnsrecord and not on the managed zone, the thing that needs it?

MZ watches secrets, changes to secret trigger MZ reconcile, changes to the state of the MZ due to the secret will trigger a DNSRecord reconcile since it owns them.

Reply:
I see you have the watch on the ManagedZone https://github.com/Kuadrant/dns-operator/pull/156/files#diff-5882be9025c1bbbef34525f6ffba174154eaac4c288cd24fb8ebb968a287bf84R173-R189, so not sure why we need this one.

Reply:
Why would the managedzone be updated due to a value in the secret changing?

Reply:
discussed on a call, will remove this.